### PR TITLE
Expose currently active file list though OCA.Files.App

### PR DESCRIPTION
**Pull request by juliusknorr**
Expose the currently active file list though OCA.Files.App to allow interaction with the current file list. Right now only OCA.Files.App.fileList (which is the same as window.FileList) are exposed but they always point to the all files list.

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: apps/files/js/app.js [hunk 1]
```diff
@@ -116,6 +118,7 @@
 					maxChunkSize: OC.appConfig.files && OC.appConfig.files.max_chunk_size
 				}
 			);
+			this.currentFileList = this.fileList
```
Comment (ChristophWurst) [inline apps/files/js/app.js]:
```suggestion
			this.updateCurrentFileList(this.fileList);
```

Does the same, but seems cleaner if we have the explicit setter :)

File: apps/files/js/app.js [hunk 2]
```diff
@@ -166,6 +169,28 @@
...
+		 *
+		 * @param {OCA.Files.FileList} newFileList
+		 */
+		updateCurrentFileList: function(newFileList) {
+			this.currentFileList = newFileList;
+		},
+
+		/**
+		 * Return the currently active file list
+		 * @return {OCA.Files.FileList}
```
Comment (ChristophWurst) [inline apps/files/js/app.js]: could also be undefined/null?
```suggestion
		 * @return {?OCA.Files.FileList}
```

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: apps/files/js/app.js (hunk 2 above)
Comment (juliushaertl) [reply apps/files/js/app.js]: Should not happen since it is set in the initialize function, but added just to be sure.

**Comment by skjnldsv** [comment]
For what do you need this?
I'm cautious to add new specific 'apis' to files as this will make it harder to migrate everything to vue

**Comment by juliushaertl** [comment]
> For what do you need this?

This would allow to trigger updates of the file list from apps even if it is not the All files view that is used. E.g. when a file is saved in Collabora we trigger a reload of the filelist in the background so that the view is updated once closed. 

> I'm cautious to add new specific 'apis' to files as this will make it harder to migrate everything to vue

Yes, this is only an intermediate step and will require apps that use it to adjust when the files app is being migrated. However those apps already rely heavily on the FileList object so those would need to be migrated anyways then.

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**